### PR TITLE
Add pallet-evm

### PR DESCRIPTION
**Pull request by haerdib**
* Pallet-evm support is added if the feature `evm` is enabled.

Related to issue https://github.com/integritee-network/worker/issues/493

### CODE REVIEW

**Comment by clangenb** [comment]
I have introduced a feature gate `evm` for this because quite some dependencies are introduced with the `evm-pallet`, so for performance reasons I want to have the option to exclude it.

**Review by clangenb — APPROVED** [review]
Looks good to me! Ready to merge after the integration tests run!

**Comment by clangenb** [comment]
@brenzi I have changed the integration tests here, so we can no longer merge because we are missing the required status checks. Can you update the rules?

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -46,9 +46,17 @@ sp-version = { default-features = false, git = "https://github.com/paritytech/su
...
 # Integritee dependencies
 pallet-parentchain = { default-features = false, git = "https://github.com/integritee-network/pallets.git", branch = "master" }
+pallet-evm = { default-features = false, optional = true, git = "https://github.com/integritee-network/frontier.git", branch = "polkadot-v0.9.24" }
 
 [features]
 default = ['std']
+# Compile the sgx-runtime with evm-pallet support in `no_std`.
+evm = ["pallet-evm"]
+# Compile the sgx-runtime with evm-pallet support in `std`.
+evm_std = [
```
Comment (haerdib) [inline runtime/Cargo.toml:56]: Is there an actual use case where the sgx-runtime will need to import a std compatible evm pallet?

Comment (clangenb) [reply runtime/Cargo.toml:56]: I can only think of `cargo test`, but I think this is worth it.

Comment (haerdib) [reply runtime/Cargo.toml:56]: Trusting you there :) Thanks for the cleanup, it looks good!

**Pull request merged**